src/terms/xc.jl

```diff
@@ -51,117 +51,41 @@ end
 end
 
-function compute_kernel(term::TermXc; ρ::RealFourierArray, ρspin=nothing, kwargs...)
-    @assert term.basis.model.spin_polarization in (:none, :spinless, :collinear)
-    density = LibxcDensity(term.basis, 0, ρ, ρspin)
-    n_spin = term.basis.model.n_spin_components
-    n_fxc  = n_spin == 1 ? 1 : 3  # Number of spin components in fxc
-    @assert 1 ≤ n_spin ≤ 2
-
-    kernel = similar(density.ρ_real, n_fxc, term.basis.fft_size...)
-    kernel .= 0
-    for xc in term.functionals
-        xc.family == :lda || error("compute_kernel only implemented for LDA")
-        terms = evaluate(xc, density; derivatives=2:2)  # only valid for LDA
-        kernel .+= terms.v2rho2
-    end
-
-    fac = term.scaling_factor
-    if n_spin == 1
-        Diagonal(vec(fac .* kernel))
-    else
-        # Blocks in the kernel matrix mapping (ρα, ρβ) ↦ (Vα, Vβ)
-        Kαα = @view kernel[1, :, :, :]
-        Kαβ = @view kernel[2, :, :, :]
-        Kβα = Kαβ
-        Kββ = @view kernel[3, :, :, :]
-
-        # Blocks in the kernel matrix mapping (ρtot, ρspin) ↦ (Vα, Vβ)
-        K_αtot  = Diagonal(vec(fac * (Kαα + Kαβ) / 2))
-        K_αspin = Diagonal(vec(fac * (Kαα - Kαβ) / 2))
-        K_βtot  = Diagonal(vec(fac * (Kβα + Kββ) / 2))
-        K_βspin = Diagonal(vec(fac * (Kβα - Kββ) / 2))
-
-        [K_αtot K_αspin;
-         K_βtot K_βspin]
-    end
-end
-
-function apply_kernel(term::TermXc, dρ::RealFourierArray, dρspin=nothing;
-                      ρ::RealFourierArray, ρspin=nothing, kwargs...)
-    basis  = term.basis
-    T      = eltype(basis)
-    n_spin = basis.model.n_spin_components
-    isempty(term.functionals) && return nothing
-    @assert all(xc.family in (:lda, :gga) for xc in term.functionals)
-    @assert basis.model.spin_polarization in (:none, :spinless, :collinear)
-
-    # Take derivatives of the density and the perturbation if needed.
-    max_ρ_derivs = maximum(max_required_derivative, term.functionals)
-    density      = LibxcDensity(basis, max_ρ_derivs, ρ, ρspin)
-    perturbation = LibxcDensity(basis, max_ρ_derivs, dρ, dρspin)
-
-    result = similar(ρ.real, basis.fft_size..., n_spin)
-    result .= 0
-    for xc in term.functionals
-        # TODO LDA actually only needs the 2nd derivatives for this ... could be optimised
-        terms = evaluate(xc, density, derivatives=1:2)
-
-        # Accumulate LDA and GGA terms in result
-        @views for σ in 1:n_spin, τ in 1:n_spin
-            στ = libxc_symmetric_index(σ, τ)
-            result[:, :, :, σ] .+= terms.v2rho2[στ, :, :, :] .* perturbation.ρ_real[τ, :, :, :]
-        end
-        if haskey(terms, :v2rhosigma)
-            @assert basis.model.spin_polarization in (:none, :spinless)
-            result[:, :, :, 1] .+= apply_kernel_gradient_correction(terms, density, perturbation)
-        end
-    end
-
-    [from_real(basis, term.scaling_factor .* result[:, :, :, σ]) for σ in 1:n_spin]
-end
+# Kernel functions given towards the end of the file
 
 #=
 The total energy is
```

Comment:
Can you make sure this big comment block is still relevant and accurate? I think we can delete the "a more algebraic way of deriving..." which was neat to convince us we were correct but is not really relevant anymore...